app/controllers/api/exchange_products_controller.rb

```diff
@@ -84,22 +85,14 @@ def render_paginated_products(paginated_products)
       )
 
       result = { products: serializer }
-      result = result.merge(pagination: pagination_data(paginated_products)) if pagination_required?
+      if pagination_required?
```

Comment:
These two places where we conditionally render pagination data (here and `Api::OrdersController`) are a bit messy, and we don't even have consistency between the two approaches. I'm wondering if we can tidy this up a bit further and either a) remove the conditional completely, or b) make it nicer and consistent/standardised.

First off I'd rename the above `serializer` variable to `serialized_products`, because that's what it is.

Secondly, I think we could try replacing all of the rest of the lines in this method with something like:
```ruby
render json: { products: serialized_products, pagination: pagination_data(paginated_products) }
```

Reply:
I think with this guard clause in place, we can also refactor the `Api::OrdersController` index action to just this:

```ruby
def index
  authorize! :admin, Spree::Order

  orders = SearchOrders.new(params, current_api_user).orders

  render json: {
    orders: serialized_orders(orders),
    pagination: pagination_data(orders)
  }
end
```